source/tutorials/installing-nixos-on-a-raspberry-pi.rst

```diff
@@ -0,0 +1,175 @@
+network interface. In case it's ``wlan0`` replace ``SSID`` and ``passphrase`` with your data and run:
+
+.. code:: shell-session 
+
+  $ wpa_supplicant -B -i wlan0 -c <(wpa_passphrase 'SSID' 'passphrase') &
+
+
+Once you see in your terminal that connection is established, run ``host google.com`` to 
+check that DNS resolves correctly.
+
+In case you've made a typo, run ``pkill wpa_supplicant`` and start over.
+
+
+Updating firmware
+-----------------
+
+To increase chances for hardware to work, we'll start by updating RPi firmware:
+
+.. code:: shell-session
+
+  $ nix-shell -p raspberrypi-eeprom
+  $ FIRMWARE_RELEASE_STATUS=stable rpi-eeprom-update -d -a
+
+  
+Installing NixOS 
+----------------
+
+For initial installation we'll install `XFCE <https://www.xfce.org/>`_ desktop environment
+with user ``guest`` and SSH daemon.
+
+.. code:: nix 
+
+  { config, pkgs, lib, ... }:
+
+  let
+    user = "guest";
+    password = "guest";
+    SSID = "mywifi";
+    SSIDpassword = "mypassword";
+    interface = "wlan0";
+    hostname = "myhostname";
+  in {
+    imports = ["${fetchTarball "https://github.com/domenkozar/nixos-hardware/archive/rpi4.tar.gz" }/raspberry-pi/4"];
+
+    fileSystems = {
+      "/" = {
+        device = "/dev/disk/by-label/NIXOS_SD";
+        fsType = "ext4";
+        options = [ "noatime" ];
+      };
+    };
+
+    networking = {
+      hostName = hostname;
+      wireless = {
+        enable = true;
+        networks."${SSID}".psk = SSIDpassword;
+        interfaces = [ interface ];
+      };
+    };
+
+    environment.systemPackages = with pkgs; [ vim ];
+
+    services.openssh.enable = true;
+
+    users = {
+      mutableUsers = false;
+      users.guest = {
+        isNormalUser = true;
+        password = password;
+        extraGroups = [ "wheel" ];
+      };
+    };
+
+    # video
+    services.xserver = {
+      enable = true;
+      displayManager.lightdm.enable = true;
+      desktopManager.xfce.enable = true;
+      videoDrivers = [ "fbdev" ];
+    };
+
+    # audio
+    hardware.pulseaudio.enable = true;
+  }
+
+To save time on typing the whole configuration, download it:
+
+.. code:: shell-session
+
+  $ curl -L https://tinyurl.com/nixos-rpi-tutorial-preview > /etc/nixos/configuration.nix 
+
+At the top of the file there are a few variables that you'd want to configure,
+most important being your wifi connection details, this time specified in declarative way.
+
+Once you're ready to install NixOS:
+
+.. code:: shell-session
+
+  $ nixos-install --root /
+  $ reboot
+
+In case your system doesn't boot, select the oldest configuration in the bootloader menu to get back to live image and start over.
+
+
+Making changes 
+--------------
+
+In case it does boot, congratulations!
+
+To make further changes to configuration, `search through NixOS options <https://search.nixos.org/options>`_,
+edit ``/etc/nixos/configuration.nix`` and update your system:
+
+.. code:: shell-session 
+
+  $ sudo su
```

Comment:
```suggestion
  $ sudo -i
```

Same concerns about this not agreeing with upstream recommendations. But now this concerns me a bit more since this is on the end-user's final system.